server/multitenancy/tenant_resolver.ts

```diff
@@ -143,7 +143,17 @@ function resolve(
     return PRIVATE_TENANT_SYMBOL;
   }
 
-  // fall back to the first tenant in the available tenants
+  /**
+   * Fall back to the first tenant in the available tenants
+   * Under the condition of enabling multitenancy, if the user has disabled both 'Global' and 'Private' tenants:
+   * it will remove the default global tenant key for custom tenant.
+   */
+  if (
+    Object.keys(availableTenantsClone).length > 1 &&
+    availableTenantsClone.hasOwnProperty(globalTenantName)
+  ) {
+    delete availableTenantsClone[globalTenantName];
```

Comment:
This will work, but it's a bit confusing why `global_tenant` is in this data structure in the first place. I went over to the security plugin to find where its coming from and it looks like its being added here: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java#L1135-L1140

I'm finding that block hard to follow and not sure what the reason for it is.

This PR looks good to me overall. I will approve when the test descriptions are updated.

Reply:
I think this `global_tenant` is basically functioning as the 'Global' tenant. Just for case that user doesn't have any custom tenants with multi-tenancy enabled but disable both 'Global' and 'Private' tenants.

Referencing to the information from @cliu123, I also left a comment in the original issue: https://github.com/opensearch-project/security-dashboards-plugin/issues/1248#issuecomment-1359959042

Reply:
@cwperks Yes, `global_tenant` comes from there.
@RyanL1997 You're right. `global_tenant` is global tenant.